squint.c

```diff
@@ -0,0 +1,1611 @@
+   111   Stack  read
+
+*/
+
+#define RI_hasD    0xc0000000 /* Destination register */
+#define RI_RdDest  0x80000000 /* Rd destination register */
+#define RI_RnDest  0x40000000 /* Rn destination register */
+
+#define RI_memOp   0x38000000 /* Mem descriptor */
+#define RI_memMask 0x30000000 /* Mem op mask */
+#define RI_memRW   0x08000000 /* Read(1)/Write(0) op */
+
+#define RI_instR   0x08000000 /* Instruction mem read */
+#define RI_dataW   0x10000000 /* Rd written to memory */
+#define RI_dataR   0x18000000 /* Rd read  from memory */
+#define RI_frameW  0x20000000 /* Rd written to frame var */
+#define RI_frameR  0x28000000 /* Rd read  from frame var */
+#define RI_push    0x30000000 /* Rd is pushed onto stack */
+#define RI_pop     0x38000000 /* Rd is popped off of stack */
+
+#define RI_bb      0x04000000 /* a basic block begins */
+#define RI_be      0x02000000 /* a basic block ends */
+#define RI_cond    0x01000000 /* indicates conditional execution */
+#define RI_branch  0x00800000 /* branch instruction */
+#define RI_func    0x00400000 /* function call */
+
+#define RI_Active  0x000000f0 /* Bitmask for register usage in inst */
+#define RI_RmAct   0x00000010
+#define RI_RsAct   0x00000020
+#define RI_RdAct   0x00000040
+#define RI_RnAct   0x00000080
+
+#define RI_Rm      0x0000000f /* register slots in ARM instruction */
+#define RI_Rs      0x00000f00
+#define RI_Rd      0x0000f000
+#define RI_Rn      0x000f0000
+
+/* NOP -- mov rn, rn */
+#define NOP       0xe1a00000
+#define NOP1      0xe1a01001
+#define NOP13     0xe1a0d00d
+
+struct ia_s {
+   int inst_addr; // inst address
+   struct ia_s *next;
+};
+
+struct pd_s {
+   int data_addr; // data address
+   struct ia_s *inst;
+} *cnst_pool;
+int cnst_pool_size;
+
+struct loop_s {
+   struct loop_s *next;
+   struct loop_s *nest; // nested loops
+   int *begin;
+   int *end;
+   int info; // stats between begin/end
+};
+
+int *cbegin;
+
+/**********************************************************/
+/*************** general utility functions ****************/
+/**********************************************************/
+
+int popcount32b(int i)
+{
+   i = i - ((i >> 1) & 0x55555555); // add pairs of bits
+   i = (i & 0x33333333) + ((i >> 2) & 0x33333333); // quads
+   i = (i + (i >> 4)) & 0x0F0F0F0F; // groups of 8
+   return (i * 0x01010101) >> 24; // horizontal sum of bytes
+}
+
+/**********************************************************/
+/********** I-stream const pool utility functions  ********/
+/**********************************************************/
+
+/* The ARM processor allows use of pc-relative constants */
+
+/* constants are sorted by the address they are stored in memory */
+int find_const(int addr)
+{
+   int low, high, mid;
+
+   // pool is sorted by *relative address* of data values
+   // with respect to beginning of text segment.
+   // binsearch pool for address of data
+   low = 0; high = cnst_pool_size;
+   while (low != high) {
+      mid = (low+high)/2;
```

Comment:
`(low + high) / 2`

(Add space between operators to match the coding style used by other lines.)